src/vmm/src/builder.rs

```diff
@@ -419,23 +419,32 @@ pub fn build_microvm_from_snapshot(
             .map_err(RestoreMicrovmState)?;
 
     // Move vcpus to their own threads and start their state machine in the 'Paused' state.
-    vmm.start_vcpus(vcpus, seccomp_filter)
-        .map_err(StartMicrovmError::Internal)?;
+    vmm.start_vcpus(vcpus, seccomp_filter).map_err(|e| {
+        let _ = vmm.exit_vcpus();
```

Comment:
I just realized that we can avoid carrying this around every error-case exit by moving it to `Vmm::drop()`.

That would also be a more complete fix; we always want to stop vcpus on dropping the vmm that owns them, right?

Reply:
Correct, I have implemented the approach you provided!